clang-15: error: builtin __has_nothrow_assign is deprecated; use __is_nothrow_assignable instead [-Werror,-Wdeprecated-builtins]

```
In file included from /home/lin/github/tiflash-1/dbms/src/IO/IOThreadPool.cpp:16:
In file included from /home/lin/github/tiflash-1/dbms/src/IO/IOThreadPool.h:17:
In file included from /home/lin/github/tiflash-1/dbms/src/Common/ThreadPool.h:18:
In file included from /home/lin/github/tiflash-1/libs/libcommon/include/boost_wrapper/priority_queue.h:22:
In file included from /home/lin/github/tiflash-1/contrib/boost/boost/heap/priority_queue.hpp:20:
In file included from /home/lin/github/tiflash-1/contrib/boost/boost/heap/detail/stable_heap.hpp:25:
In file included from /home/lin/github/tiflash-1/contrib/boost/boost/type_traits/is_nothrow_move_assignable.hpp:16:
/home/lin/github/tiflash-1/contrib/boost/boost/type_traits/has_nothrow_assign.hpp:65:7: error: builtin __has_nothrow_assign is deprecated; use __is_nothrow_assignable instead [-Werror,-Wdeprecated-builtins]
      BOOST_HAS_NOTHROW_ASSIGN(T)
      ^
/home/lin/github/tiflash-1/contrib/boost/boost/type_traits/intrinsics.hpp:205:43: note: expanded from macro 'BOOST_HAS_NOTHROW_ASSIGN'
#     define BOOST_HAS_NOTHROW_ASSIGN(T) (__has_nothrow_assign(T) && !is_volatile<T>::value && is_assignable<T&, const T&>::value)
                                          ^
1 error generated.
```

--- a/libs/libcommon/include/boost_wrapper/priority_queue.h
+++ b/libs/libcommon/include/boost_wrapper/priority_queue.h
@@ -16,7 +16,9 @@
 
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-parameter"
-
+#if defined(__clang__) && __clang_major__ >= 15
+#pragma GCC diagnostic ignored "-Wdeprecated-builtins"
+#endif
 #include <boost/heap/priority_queue.hpp>
 
 #pragma GCC diagnostic pop

